### PR TITLE
Adds option to settings page to show stats for different roles

### LINKED ISSUE #106: Add option on settings page to show stats for different roles
A filter is sometimes scary for beginners. Why not adding this as an option on the settings page?
https://github.com/pluginkollektiv/statify/wiki/en-Hooks#statify__user_can_see_stats

Are there any reasons why this should not be done?

### PR DESCRIPTION
**Pull request by zottto**
Fixes #106.

### CODE REVIEW

**Comment by florianbrinkmann** [comment]
The PHP Code Style Check detected a few issues https://github.com/pluginkollektiv/statify/actions/runs/4460531084/jobs/7833735004?pr=248

**Comment by florianbrinkmann** [comment]
Besides that: LGTM

File: inc/class-statify-dashboard.php [hunk 1]
```diff
@@ -37,13 +37,24 @@ class Statify_Dashboard extends Statify {
...
-		if ( ! apply_filters( 'statify__user_can_see_stats', current_user_can( 'edit_dashboard' ) ) ) {
-			return;
+		// Backwards compatibility for older statify versions without this option.
+		if ( ! isset( Statify::$_options['show_widget_roles'] ) ) {
+			// Filter user_can_see_stats.
+			$can_edit = apply_filters( 'statify__user_can_see_stats', current_user_can( 'edit_dashboard' ) );
+		} else {
+			$statify_roles = Statify::$_options['show_widget_roles'];
+			$current_user = wp_get_current_user();
+			$user_roles = (array) $current_user->roles;
```
Comment (stklcode) [inline inc/class-statify-dashboard.php:47]: Is there any point in casting `$current_user` to `array` while using object operator `->`?

**Comment by stklcode** [comment]
Rebased onto _develop_ and resolved merge conflicts (logic has been extracted into a different method recently).

Just pointing out one thing:
The new settings option also implies that all users who are capable of seeing the widget are also allowed to edit it.
(this is not new and might be targeted in a different issue)

**Review by stklcode — APPROVED** [review]
Resolved all merge-conflicts and added a unit test for the new behavior.

LGTM

**Pull request merged**